gazebosim/gz-sim
# Add Lens Flare System

# 🎉 New feature

Closes #1909 

## Summary
 - Added Lens Flare System that adds Lens Flare Render Pass to the camera and allows users to enable lens flare in camera images. The lens flares can be configured using `<scale>`, `<color>`, and `<occlusion_steps>` tags in the SDF.
 - Added an example `camera_lens_flare.sdf` file for testing which contains 2 cameras(wide angle and normal) with different configurations for the lens flare (one default and other custom).

![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/23265149/225682546-8293595a-f9c2-48a8-966d-1dcc7380541f.gif)

## Test it
 - Make sure to use the `gz-sim8` and compatible versions of other packages especially `gz-rendering8` since it contains the Lens Flare Pass.
 - Launch the `examples/worlds/camera_lens_flare.sdf` file from the `gz-sim` repo using:
 ```bash
gz sim camera_lens_flare.sdf
```

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.



# 🎈 Release

Preparation for <X.Y.Z> release.

Comparison to <x.y.z>: https://github.com/gazebosim/<REPO>/compare/<LATEST_TAG_BRANCH>...<RELEASE_BRANCH>


Needed by <PR(s)>

## Checklist
- [ ] Asked team if this is a good time for a release
- [ ] There are no changes to be ported from the previous major version
- [ ] No PRs targeted at this major version are close to getting in
- [ ] Bumped minor for new features, patch for bug fixes
- [ ] Link to PR updating dependency versions in appropriate repository in [gazebo-release](https://github.com/gazebo-release) (as needed): <LINK>



**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.
